tests/FsCheck.Test/Gen.fs

```diff
@@ -120,6 +120,13 @@ module Gen =
         |> sample1
         |> ((=) (List.init length (fun _ -> v)))
 
+    [<Property>]
+    let ``shuffle is isomorphic`` (xs : int array) =
+        Gen.shuffle xs
+        |> sample1
+        |> Array.sort
+        |> ((=) (Array.sort xs))
+
```

Comment:
:+1: 

It looks indeed like both input and output are isomorphic, given than both are of the same form and shape, and there's a structure-preserving one-to-one correspondence between them.